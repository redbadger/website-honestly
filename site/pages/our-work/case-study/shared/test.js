// @flow

import React from 'react';
import { shallow } from 'enzyme';

import {
  PageHeading,
  Body,
  Quote,
  Section,
  SectionHeading,
  SectionBody,
  H3,
  P,
  UL,
  ArticleImg,
} from '.';

const render = component => shallow(component);

describe('pages/our-work/case-study/shared', () => {
  it('renders PageHeading correctly', () => {
    expect(render(<PageHeading>Heading</PageHeading>)).toMatchSnapshot();
  });

  it('renders Body correctly', () => {
    expect(
      render(
        <Body>
          <PageHeading>Heading</PageHeading>
        </Body>,
      ),
    ).toMatchSnapshot();
  });

  it('renders Quote correctly', () => {
    expect(
      render(<Quote author={{ name: 'James', title: 'Big Boss' }} text="Great job" />),
    ).toMatchSnapshot();
  });

  it('renders Section correctly', () => {
    expect(
      render(
        <Section>
          <SectionBody>Body</SectionBody>
        </Section>,
      ),
    ).toMatchSnapshot();
  });

  it('renders SectionHeading correctly', () => {
    expect(
      render(<SectionHeading subHeading="I am subheading" heading="I am heading" />),
    ).toMatchSnapshot();
  });

  it('renders SectionBody correctly', () => {
    expect(render(<SectionBody>Body</SectionBody>)).toMatchSnapshot();
  });

  it('renders P correctly', () => {
    expect(render(<P>Hey you</P>)).toMatchSnapshot();
  });

  it('renders UL correctly', () => {
    expect(
      render(
        <UL>
          <li>List item</li>
        </UL>,
      ),
    ).toMatchSnapshot();
  });

  it('renders ArticleImg correctly', () => {
    expect(render(<ArticleImg src="/path/to/image" alt="A sunset" />)).toMatchSnapshot();
  });

  it('renders H3 correctly', () => {
    expect(render(<H3>heading</H3>)).toMatchSnapshot();
  });
});
