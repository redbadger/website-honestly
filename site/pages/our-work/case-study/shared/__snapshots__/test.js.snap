// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/our-work/case-study/shared renders ArticleImg correctly 1`] = `
<img
  alt="A sunset"
  className="articleImg"
  src="/path/to/image"
/>
`;

exports[`pages/our-work/case-study/shared renders Body correctly 1`] = `
<div
  className="body"
>
  <PageHeading>
    Heading
  </PageHeading>
</div>
`;

exports[`pages/our-work/case-study/shared renders H3 correctly 1`] = `
<H3
  type="fontS2"
>
  heading
</H3>
`;

exports[`pages/our-work/case-study/shared renders P correctly 1`] = `
<P
  customClass="paragraph"
>
  Hey you
</P>
`;

exports[`pages/our-work/case-study/shared renders PageHeading correctly 1`] = `
<div
  className="heading"
>
  <H1
    type="fontL"
  >
    Heading
  </H1>
</div>
`;

exports[`pages/our-work/case-study/shared renders Quote correctly 1`] = `
<Quote
  author={
    Object {
      "name": "James",
      "title": "Big Boss",
    }
  }
  className="quote"
  text="Great job"
/>
`;

exports[`pages/our-work/case-study/shared renders Section correctly 1`] = `
<div
  className="section"
>
  <SectionBody>
    Body
  </SectionBody>
</div>
`;

exports[`pages/our-work/case-study/shared renders SectionBody correctly 1`] = `
<div
  className="sectionBody"
>
  Body
</div>
`;

exports[`pages/our-work/case-study/shared renders SectionHeading correctly 1`] = `
<h2
  className="sectionHeading"
>
  <span
    className="redTitle"
  >
    I am subheading
  </span>
  I am heading
</h2>
`;

exports[`pages/our-work/case-study/shared renders UL correctly 1`] = `
<ul
  className="list"
>
  <li>
    List item
  </li>
</ul>
`;
